pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,995 @@
+        """
+        group = 'acceptance_fraction'
+        if walkers is None:
+            wmask = numpy.ones(fp.nwalkers, dtype=bool)
+        else:
+            wmask = numpy.zeros(fp.nwalkers, dtype=bool)
+            wmask[walkers] = True
+        return fp[group][wmask]
+
+    def write_results(self, fp, max_iterations=None):
+        """Writes metadata, samples, likelihood stats, and acceptance fraction
+        to the given file. See the write function for each of those for
+        details.
+
+        Parameters
+        -----------
+        fp : InferenceFile
+            A file handler to an open inference file.
+        max_iterations : {None, int}
+            If results have not previously been written to the
+            file, new datasets will be created. By default, the size of these
+            datasets will be whatever the length of the sampler's chain is at
+            this point. If you intend to run more iterations in the future,
+            set this value to that size so that the array in the file will be
+            large enough to accomodate future data.
+        """
+        self.write_metadata(fp)
+        self.write_chain(fp, max_iterations=max_iterations)
+        self.write_likelihood_stats(fp, max_iterations=max_iterations)
+        self.write_acceptance_fraction(fp)
+
+
+class EmceePTSampler(_BaseMCMCSampler):
+    """This class is used to construct a parallel-tempered MCMC sampler from
+    the emcee package's PTSampler.
+
+    Parameters
+    ----------
+    likelihood_evaluator : likelihood class
+        An instance of the likelihood class from the
+        pycbc.inference.likelihood module.
+    ntemps : int
+        Number of temeratures to use in the sampler.
+    nwalkers : int
+        Number of walkers to use in sampler.
+    processes : {None, int}
+        Number of processes to use with multiprocessing. If None, all available
+        cores are used.
+    burn_in_iterations : {None, int}
+        Set the number of burn in iterations to use. If None,
+        `burn_in_ieterations` will be initialized to 0.
+    """
+    name = "emcee_pt"
+
+    def __init__(self, likelihood_evaluator, ntemps, nwalkers, processes=None,
+                 burn_in_iterations=None):
+
+        try:
+            import emcee
+        except ImportError:
+            raise ImportError("emcee is not installed.")
+
+        # construct the sampler: PTSampler needs the likelihood and prior
+        # functions separately
+        likelihood_evaluator.set_callfunc('loglikelihood')
+
+        # initialize the pool to use
+        if processes == 1:
+            pool = None
+        else:
+            pool = emcee.interruptible_pool.InterruptiblePool(
+                processes=processes)
+
+        ndim = len(likelihood_evaluator.waveform_generator.variable_args)
+        sampler = emcee.PTSampler(ntemps, nwalkers, ndim,
+                                  likelihood_evaluator,
+				  likelihood_evaluator._prior,
```

Comment:
__[flake8]__
* [E101] indentation contains mixed spaces and tabs
* [W191] indentation contains tabs
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->